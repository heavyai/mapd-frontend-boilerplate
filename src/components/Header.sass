.container
  display: flex
  align-items: center
  box-sizing: border-box
  height: 64px
  padding: 12px
  background-color: #282C34

.logo
  width: 53px

.tabBar
  \:global(.mdc-tab-bar__indicator)
    background-color: #a0f
