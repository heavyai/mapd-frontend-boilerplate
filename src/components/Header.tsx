import React from 'react'
import Elevation from 'rmwc/Elevation'
import { TabBar, Tab } from 'rmwc/Tabs'

import Logo from '~/assets/logo.svg'
import { setActiveTabIndex } from '~/actions/tabs'

const styles = require( './Header.sass' )


// Types
import { Dispatcher } from '~/actions'
import { TabsState } from '~/reducers/tabs'

type Props = {
  dispatch: Dispatcher,
  tabs: TabsState
}


const Header: React.StatelessComponent<Props> = ( { dispatch, tabs: { activeTabIndex } } ) => {
  const onChangeTab = ( event: React.FormEvent<HTMLInputElement> ) =>
    dispatch( setActiveTabIndex( Number( event.currentTarget.value ) ) )

  return (
    <Elevation z={ 1 } className={ styles.container }>
      <Logo className={ styles.logo } />
      <TabBar className={ styles.tabBar } activeTabIndex={ activeTabIndex } onChange={ onChangeTab }>
        <Tab>SQL Editor</Tab>
        <Tab>Chart</Tab>
      </TabBar>
    </Elevation>
  )
}


export default Header
