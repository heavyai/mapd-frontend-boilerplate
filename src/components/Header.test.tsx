import React from 'react'
import { render } from 'enzyme'

import Header from './Header'


test( 'Component renders', () => {
  const component = render(
    <Header dispatch={ jest.fn() } tabs={ { activeTabIndex: 0 } } />
  )

  expect( component ).toMatchSnapshot()
} )
