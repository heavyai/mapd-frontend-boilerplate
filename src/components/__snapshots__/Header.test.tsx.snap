// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
<div
  class="container  mdc-elevation--z1"
>
  <div />
  <nav
    class="tabBar  mdc-tab-bar"
  >
    <div
      class=" mdc-tab"
    >
      SQL Editor
    </div>
    <div
      class=" mdc-tab"
    >
      Chart
    </div>
    <span
      class=" mdc-tab-bar__indicator"
    />
  </nav>
</div>
`;
